subprojects/base-services/src/main/java/org/gradle/util/GFileUtils.java

```diff
@@ -302,7 +302,7 @@ public static void mkdirs(File dir) {
         }
 
         if (dir.exists() && !dir.isDirectory()) {
-            throw new UncheckedIOException(String.format("Cannot create directory '%s' as it already exists, but is not a directory", dir));
+            throw new UncheckedIOException(String.format("Cannot create directory '%s' as it already exists, but is not a directory", dir), new IOException());
```

Comment:
I don't think creating a message-less `IOException` is a good solution, or if this makes things better.

If you really want to go this way `throw new UncheckedIOException(new IOException("..."))` would be a more correct route, but it would still not be a worthwhile improvement over what we have already.

I think we should revert this commit.

Reply:
> throw new UncheckedIOException(new IOException("..."))

this would have a weird message, tried it before

Reply:
IIUC this has been reverted.